### PR TITLE
Handle friendly url automatic update and automatic offline shop configurations

### LINKED ISSUE #33670: [BO] PPV2  Friendly URL is not updated automatically after updating a product name
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When enabling the Force update of friendly URL in shop parameters > Product Settings , the Friendly URL is not updated automatically after updating a product name 
 

### Expected behavior

We only update the URL field content in the frontend thanks to JS.
On name input change the value from name input is sluggified and automatically put in the friendly URL input.

We only do this automatic change when the product is **offline** to avoid breaking production URLs.

Label in configuration form: When enabled, friendly URL will be updated on every name change for offline product only

The button `Reset URL` should be renamed into `Generate URL from name` and it should be enabled when the name is modified and the URL doesn't match anymore

### Steps to reproduce

1. Go to 'Shop parameters > Product Settings' page
2. Enable force update friendly URL
3. Go to product page and create a product > save 
4. Update name of  the created product > name of the product is update 
==> Friendly URL should updated automatically

https://github.com/PrestaShop/PrestaShop/assets/31663369/7d140dae-f910-4297-9db9-43aae1288d59

### PrestaShop version(s) where the bug happened

PS 81x

### PHP version(s) where the bug happened

PHP 8.1.0

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_

### LINKED ISSUE #33671: [BO] PPV2 Enable, disable default product status is not applied automatically
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When enabling the option **Activate new products by default** in shop parameters > Product Settings , and try to create a new product the status is alwyas OFFLINE . (it should be online by default )

### Expected behavior

The default product status should be Online. The product can only be save online if the product name is fulfilled. 

### Steps to reproduce

1. Go to 'Shop parameters > Product Settings' page
2. Enable Activate new products by default > save 
3. Go to product page and create a product 
=>> The product status is OFFLINE 

https://github.com/PrestaShop/PrestaShop/assets/31663369/7054a388-1977-48cb-b971-b90c6e70f426

### PrestaShop version(s) where the bug happened

PS 8.1.x

### PHP version(s) where the bug happened

php 810

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Handle friendly url automatic update and automatic offline shop configurations in the new product page
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Check the issues description for expected behaviour UI tests https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/6081917691
| Fixed issue or discussion?     | Fixes #33670 and Fixes #33671
| Related PRs       | ~
| Sponsor company   | ~

### CODE REVIEW

File: src/PrestaShopBundle/Form/Admin/Sell/Product/SEO/SEOType.php [hunk 1]
```diff
@@ -78,20 +84,28 @@ public function __construct(
...
         $this->legacyContext = $legacyContext;
+        $this->configuration = $configuration;
     }
 
     /**
      * {@inheritdoc}
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        // Automatic update is only enabled when product is offline and the configuration is enabled
```
Comment (PululuK) [inline src/PrestaShopBundle/Form/Admin/Sell/Product/SEO/SEOType.php:103]: Should be a catalog config var, something like `PS_FORCE_FRIENDLY_PRODUCT_ACTIVE_ONLY` ?

should we also take into account the fields like `visibility`, `on_sale`, and `available_for_order` ?

**Review by tleon — APPROVED** [review]
LGTM.

**Review by M0rgan01 — APPROVED** [review]

**Review by boherm — APPROVED** [review]

**Review by MhiriFaten — COMMENTED** [review]
Hello @jolelievre ,

I have checked your PR and the issue of **_Enable, disable default product status_** is well fixed, but I noticed that the issue of friendly URL still exists.

[Untitled_ Sep 5, 2023 11_06 AM.webm](https://github.com/PrestaShop/PrestaShop/assets/111756615/148eea5c-b7e8-4efd-b748-08347eb6ed12)

Could you check it please?

**Comment by jolelievre** [comment]
Hi @MhiriFaten 

for the friendly URL considering your video it's on purpose, the automatic update is only performed on Offline products, you can see it in the expected behaviour in the issue:

> We only do this automatic change when the product is offline to avoid breaking production URLs.

But it's indeed different from the old page where it was systematic, you can test to turn your product offline then the automatic update will be effective (you can still force a different value manually though, that's also on purpose).

**Review by MhiriFaten — APPROVED** [review]
@jolelievre thanks for the explanation,
LGTM :heavy_check_mark:

**Comment by kpodemski** [comment]
thank you @jolelievre @MhiriFaten

**Pull request merged**